bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,317 @@
+        - "config"
+      summary: "Get kubeflow config entry file"
+      security:
+        - BearerToken: []
+      operationId: "getConfigFile"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "config"
+          description: "config entry name"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "version"
+          description: "config entry kubeflow version"
+          required: true
+          type: "string"
+        - in: "query"
+          name: "filepath"
+          description: "file relative path"
+          required: true
+          type: "string"
+      responses:
+        200:
+          description: "Kubeflow config entries"
+          schema:
+            $ref: "#/definitions/KfConfigFile"
+        404:
+          description: "Entry not found"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+    put:
+      tags:
+        - "config"
+      description: "Update kubeflow config entry file"
+      security:
+        - BearerToken: []
+      operationId: "putConfigFile"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "config"
+          description: "config entry name"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "version"
+          description: "config entry kubeflow version"
+          required: true
+          type: "string"
+        - in: "body"
+          name: "body"
+          description: "file object to update"
+          required: true
+          schema:
+            $ref: "#/definitions/KfConfigFile"
+      responses:
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+  /projects/{project}/deployments:
+    post:
+      tags:
+        - "deployment"
+      summary: "Create a new kubeflow deployment on GCP"
+      security:
+        - BearerToken: []
+      operationId: "createDeployment"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "body"
+          name: "body"
+          description: "Deployment name"
+          required: true
+          schema:
+            $ref: "#/definitions/KfDeployment"
```

Comment:
How does kfDeployment relate to the Config?

Is KfDeployment just a pointer to the config?

Reply:
Relation between kfDeployment and KfConfig is like
relation between notebook instance and notebook image:
User can spawn up kfDeployment from existing KfConfig entries.